pkg/workflow/task/workload/create.go

```diff
@@ -1,31 +1,102 @@
 	"fmt"
+	"io"
 
 	"github.com/aws/eks-anywhere/pkg/types"
 	"github.com/aws/eks-anywhere/pkg/workflow/workflowcontext"
 )
 
+// Cluster represents a workload cluster to be created.
+type Cluster interface {
+	// Generate and apply the cluster manifest.
+	ApplyManifest(_ context.Context, management *types.Cluster) error
+
+	// GetName retrieves the cluster name.
+	GetName() string
+
+	// WriteKubeconfig writes the Kuberconfig for this cluster to the io.Writer.
+	WriteKubeconfig(io.Writer) error
+
+	// NotifyFirstControlPlaneReady signals on the injected channel when the first control plane
+	// is ready. The node is ready when its possible to interact with the Kube API server using
+	// a Kubeconfig.
+	NotifyFirstControlPlaneReady(_ context.Context, management *types.Cluster, _ <-chan struct{})
+
+	// NotifyReady signals on the injected channel when all cluster nodes are ready. All nodes
+	// are ready when its possible to deploy workloads to the cluster.
+	NotifyReady(_ context.Context, management *types.Cluster, _ <-chan struct{})
+}
+
 // CNIInstaller install a CNI in a given cluster.
 type CNIInstaller interface {
 	// Install configures a CNI for the first time in a kubernetes cluster
 	Install(ctx context.Context, cluster *types.Cluster) error
 }
 
-// Create creates a Kubernetes conformant cluster.
+// Create creates a Kubernetes conformant cluster that is immediately usable for simple workloads.
+// It expects a management cluster configuration to be available in the context.
 type Create struct {
-	CNI CNIInstaller
+	CNI     CNIInstaller
+	Cluster Cluster
 }
 
+/*
+
+Things to care about
+
+CloudStack team feels CAPI MachineHealthChecks are dependent on CNI installation to turn healthy.
+
+We think CAPI will use MachineHealthChecks to detect unhealthy nodes during cluster provisioning
+and if one is unhealthy, it will remediate.
+
+Might need to validate MachineHealthChecks are necessary for initial cluster provisioning.
+
+How do we unwind all these actions?
+
+Need to write the CAPI objects we generate to the disk. This is presumeably for debugging
+purposes.
+
+*/
+
 // RunTask satisfies workflow.Task.
 func (t Create) RunTask(ctx context.Context) (context.Context, error) {
-	// TODO: add provider create cluster, kubeconfig retrieve and write to disk
+	managementCluster := workflowcontext.ManagementCluster(ctx)
+	if managementCluster == nil {
+		return nil, fmt.Errorf("no management cluster in context")
+	}
+
+	// TODO(chrisdoherty) Determine a better name. This name implies CAPI stuff and while we aren't
+	// strictly trying to hide the use of CAPI we should refrain, ideally, from limiting the API
+	// to CAPI manifest generation and application only.
+	if err := t.Cluster.ApplyManifest(ctx, managementCluster); err != nil {
+		return nil, err
+	}
+
+	firstControlPlaneReady := make(chan struct{})
+	t.Cluster.NotifyFirstControlPlaneReady(ctx, managementCluster, firstControlPlaneReady)
+	select {
+	case <-ctx.Done():
+		// Throw error
```

Comment:
are you thinking about adding a way for the cluster to notify of an error?

Reply:
The channel will probably be `<-chan error` and we'll give `nil` semantics.